internal/controller/instaslice_daemonset.go

```diff
@@ -783,6 +593,16 @@ func (r *InstaSliceDaemonsetReconciler) SetupWithManager(mgr ctrl.Manager) error
 			}
 		}
 
+		errRetrievingInstaSlicePostSetup := r.Get(ctx, typeNamespacedName, &instaslice)
+		if errRetrievingInstaSlicePostSetup != nil {
+			log.FromContext(ctx).Error(errRetrievingInstaSlicePostSetup, "unable to fetch InstaSlice resource for node")
+			os.Exit(1)
```

Comment:
We should avoid doing os.Exit(1) in daemonsets... Is there a way to avoid the os.Exit(1)?

Reply:
System cannot function without InstaSlice CR, this to me is a fatal error

Reply:
Would it make sense to return the err instead of abruptly existing here? Also like other places, you can simply use `err` for the variable name. 

Instead of allowing `os.Exit` from multiple locations, you can return the err and the caller can gracefully take the action.

Reply:
This is the setup call when the manager is setup for daemonset, The main function calls this setup function, if the GPU is unhealthy hence the setup is failing then should we bother to even retry?

Reply:
Thanks, I have removed os.Exit